aws/cloudformation/ami-manager.js

```diff
@@ -6,42 +6,102 @@
 // This module is automatically provided to ZipFile-based Lambda functions.
 // Ref: http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#cfn-lambda-function-code-cfnresponsemodule
```

Comment:
Summary of `ami-manager.js` changes:

- lots more edge-case and error-handling
- wait until the ec2 instance is in 'stoppped' state before starting AMI creation
- support waiting longer than 5 minutes (the maximum amount of time for a single Lambda-function execution) for AMI creation by using a 'recursive function' calling strategy (wait until just before the function times-out, then call itself using `lambda.invoke`, passing along all of the event parameters).